pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java

```diff
@@ -130,6 +140,19 @@ public Object handle(Request request, Response response, UserRegistrationPayload
                         auth0ClientSecret, payload.getRedirectUri());
                 auth0UserId.set(Auth0Util.getVerifiedAuth0UserId(auth0RefreshResponse.getIdToken(), auth0Domain));
                 LOG.info("Successfully exchanged auth0 code for auth0UserId {} for local registration", auth0UserId);
+
+                // Look for potential user metadata.
+                var auth0User = auth0Util.getAuth0User(auth0UserId.get(), mgmtClient.getToken());
```

Comment:
I believe we should be checking if (payload.getFirstName() == null || payload.getLastName() == null) before making Auth0 call to avoid unnecessary Auth0 call if this data is already passed in payload.

Reply:
Good call.

Reply:
Actually, I'm deciding not to add the check. In local registration flow, we're almost always not going to get name in the payload, so we're almost always making the call.

Reply:
If we do end up passing first/last name in the payload, there's going to be other (real production) code changes required as well. Otherwise there will be a disparity between real registration vs local regstration. I don't see that happening anytime soon.